databricks_cli/execution_context/cli.py

```diff
@@ -0,0 +1,251 @@
+              help="The language for the context.")
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def create_context_cli(api_client, cluster_id, language): #  NOQA
+    """
+    Creates a new execution context.
+    """
+    result = ExecutionContextApi(api_client).create_context(cluster_id, language)
+    click.echo(pretty_format(result))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help="Gets information about an execution context.")
+@click.option('--cluster-id', required=True, type=ClusterIdClickType(),
+              help=ClusterIdClickType.help)
+@click.option('--context-id', required=True, type=ExecutionContextIdClickType(),
+              help=ExecutionContextIdClickType.help)
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def get_context_status_cli(api_client, cluster_id, context_id): #  NOQA
+    """
+    Gets information about an execution context.
+    """
+    result = ExecutionContextApi(api_client).get_context_status(cluster_id, context_id)
+    click.echo(pretty_format(result))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help="Deletes an execution context.")
+@click.option('--cluster-id', required=True, type=ClusterIdClickType(),
+              help=ClusterIdClickType.help)
+@click.option('--context-id', required=True, type=ExecutionContextIdClickType(),
+              help=ExecutionContextIdClickType.help)
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def delete_context_cli(api_client, cluster_id, context_id): #  NOQA
+    """
+    Deletes an execution context.
+    """
+    result = ExecutionContextApi(api_client).delete_context(cluster_id, context_id)
+    click.echo(pretty_format(result))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help="Executes a command within an existing execution context.")
+@click.option('--output', default=None, help=CommandOutputType.help, type=CommandOutputType())
+@click.option('--cluster-id', required=True, type=ClusterIdClickType(),
+              help=ClusterIdClickType.help)
+@click.option('--context-id', required=True, type=ExecutionContextIdClickType(),
+              help=ExecutionContextIdClickType.help)
+@click.option('--command', required=True, type=CommandStringType(),
+              help=CommandStringType.help)
+@click.option('--wait', required=False, default=False, type=click.BOOL,
+              help="If true wait for command completion, otherwise schedule "
+                   "the command and return immediately.")
+@click.option('--language', required=False, default="python", 
+              type=click.Choice(['python', 'scala', 'sql'], case_sensitive=False),
+              help="The language for the context.")
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def execute_command_cli(api_client, output, cluster_id, context_id, command, language, wait): #  NOQA
```

Comment:
from end-user perspective, this command is better be exposed under "clusters" command group. e.g. `databricks cluster command this_thing.py --cluster-name "Shared Autoscaling"`. Language could be parsed from file argument extension (`this_thing.py`). In most of the cases, users don't want to copy-paste cluster_id and they want to supply names instead. If cluster is not running - start it. also need to thing about stopping cluster after command is finished, if cluster was not running before that.

Reply:
this command actually fixes this issue - https://github.com/databricks/databricks-cli/issues/341 - and solves certain operational workflow.